src/Gir/Generation/Callback.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+namespace Gir
+{
+	public partial class Callback : IGeneratable
+	{
+		public void Generate (GenerationOptions opts)
+		{
+			using (var writer = this.GetWriter (opts))
+			{
+				writer.WriteLine ("namespace " + opts.Namespace.Name);
+				writer.WriteLine ("{");
+				using (writer.Indent ()) {
+					// TODO: Validation if we have an userdata parameter we can use so we can write this code.
+					// Otherwise, we need to generate code which uses a non-static callback.
+
+					var returnType = GetReturnType (writer);
+					
+					// Public API delegate which uses managed types.
+					writer.WriteLine ("[UnmanagedFunctionPointer (CallingConvention.Cdecl)]");
+					writer.WriteLine ($"public delegate {returnType} {Name} PARAMS");
+					writer.WriteLine ();
```

Comment:
Why not use https://github.com/mono/gir-sharp/blob/master/src/Gir/Generation/IGeneratableExtensions.cs#L100 ?

Reply:
I wrote this for now, need to hack out the `import` definition out of `ICallable`

Reply:
And a callback is not really a member, it's a type.